### ISSUE TITLE
'hotfix branch from Release' inconsistent with other Microsoft guidance

### ISSUE THREAD

**Issue by SCordesHum** [MicrosoftDocs/vsts-docs#2070]
Several other Microsoft Release Flow documentation pages describe hotfixes as:  create a feature branch from master, merge to master, and cherry-pick from master into the Release branch(es).

This page says to create the hotfix branch from the Release branch, merge the hotfix branch to the Release branch, cherry-pick from the Release branch to a feature branch off of master, and merge that feature branch to master.

Since (1) the rationale for branching from master is solid (i.e., so that you don't miss putting the fix in future releases), and (2) if I understand correctly, branching from master eliminates one of the branches (i.e., the feature branch IS the hotfix branch), that seems like the better approach.

So please update this document to make give consistent guidance.  Or at the very least, indicate that this document is showing an alternative approach and tell why you would choose it instead.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: fe168fc3-2ed2-b3e6-ce30-cac721165ceb
* Version Independent ID: 8d63fbd7-12c4-9a94-92f6-969e7eb41037
* Content: [Git branching guidance - Azure Repos](https://docs.microsoft.com/en-us/azure/devops/repos/git/git-branching-guidance?view=vsts)
* Content Source: [docs/repos/git/git-branching-guidance.md](https://github.com/MicrosoftDocs/vsts-docs/blob/master/docs/repos/git/git-branching-guidance.md)
* Product: **devops**
* GitHub Login: @steved0x
* Microsoft Alias: **sdanie**

**Comment by stoopered** [comment]
It's now Feb 2019, and this document still has not been updated...

**Comment by Victor-Savu** [comment]
My (unsolicited) 2 cents: do the most bespoke fix you can for the code that is released. Afterwards relax, take a deep breath, and then think about what may need to be done for the master branch.

The fix may or may not be relevant for the `master` branch due to the changes that is has received since the release. The current release definitely needs the fix (it needed it the day before the release, but.. well.. here we are!). So, by default, the fix should be performed based on the release branch that needs it and then the solution may be ported to the master branch as appropriate.

Only in rare cases does porting the fix involve straight-up cherry-picking or merging. Often times, the problem doesn't even manifest in master anymore (problems tend to lie in pieces of code that are more recent, more subject to change and likely to be replaced even before all of their bugs get to manifest). Even when the porting involves cherry-picking or merging, more often than not, some conflict solving would be involved, with some other changes on top for the alien code to fit the new.

**Comment by Chaithanya-Bangaru** [comment]
create a feature branch from master, merge to master, and cherry-pick from master into the Release branch(es).

First paragraph is Theoritical lesson., 


Second paragraph is Practical lesson.
create the hotfix branch from the Release branch, merge the hotfix branch to the Release branch, cherry-pick from the Release branch to a feature branch off of master, and merge that feature branch to master

**Comment by luisgmgouveia** [comment]
I agree this page is not consistent with the 'Release Flow' proposed by Microsoft. According to that model, we should never merge the release branches back to master.

**Comment by o-o00o-o** [comment]
Following "Release Flow" guidance we have had a case where unintended changes were included in the Cherry Pick. This was because the developer did a merge from master in his "feature\fix" branch. This seems to then include all the master changes into the feature and then when the CP is done it brings in all the changes from master at that point.

However with the solution given in this document (to fix in release and back port) this has the risk that the change is forgotten to make it into master and next release you have a regression.

In my experience on the projects I've worked on, fixes found during release usually do need to be put into master, so the Release Flow method (CP from master commit to bring in fix to Release branch) is better as there is no chance of forgetting to back-port
